src/main/java/org/wildfly/security/audit/SyslogAuditEndpoint.java

```diff
@@ -0,0 +1,160 @@
+
+        synchronized(this) {
+            if (!accepting) return;
+
+            syslogHandler.doPublish(new ExtLogRecord(toLevel(t), u, SyslogAuditEndpoint.class.getName()));
+        }
+    }
+
+    private static Level toLevel(EventPriority eventPriority) {
+        switch (eventPriority) {
+            case ALERT:
+            case EMERGENCY:
+            case CRITICAL:
+            case ERROR:
+                return Level.SEVERE;
+            case WARNING:
+                return Level.WARNING;
+            case INFORMATIONAL:
+                return Level.INFO;
+            default:
+                return Level.FINEST;
+        }
+    }
+
+    @Override
+    public void close() throws IOException {
+        accepting = false;
+
+        synchronized(this) {
+            syslogHandler.close();
+        }
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+
+        private InetAddress serverAddress;
+        private int port;
+        private boolean tcp = true;
+        private String hostName;
+
+        Builder() {
+        }
+
+        /**
+         * Set the server address syslog messages should be sent to.
+         *
+         * @param serverAddress the server address syslog messages should be sent to.
+         * @return this builder.
+         */
+        public Builder setServerAddress(InetAddress serverAddress) {
+            this.serverAddress = checkNotNullParam("serverAddress", serverAddress);
+
+            return this;
+        }
+
+        /**
+         * Set the port the syslog server is listening on.
+         *
+         * @param port the port the syslog server is listening on.
+         * @return this builder.
+         */
+        public Builder setPort(int port) {
+            this.port = port;
+
+            return this;
+        }
+
+        /**
+         * Set if the communication should be using TCP.
+         *
+         * @param tcp if the communication should be using TCP.
+         * @return this builder.
+         */
+        public Builder setTcp(boolean tcp) {
+            this.tcp = tcp;
+
+            return this;
+        }
+
+        /**
+         * Set the host name that should be sent within the syslog messages.
+         *
+         * @param hostName the host name that should be sent within the syslog messages.
+         * @return this builder.
+         */
+        public Builder setHostName(String hostName) {
+            this.hostName = checkNotNullParam("hostName", hostName);
+
+            return this;
+        }
+
+        /**
+         * Build a new {@link AuditEndpoint} configured to pass all messages using Syslog.
+         *
+         * @return a new {@link AuditEndpoint} configured to pass all messages using Syslog.
+         * @throws IOException if an error occurs initialising the endpoint.
+         */
+        public AuditEndpoint build() throws IOException {
+            return new SyslogAuditEndpoint(this);
```

Comment:
Seems like there ought to be some protection from creating counterfeit audit loggers, though I can't think of a fool-proof mechanism that can do so.